aio: title and search index fixes

petebacondarwin opened the pull request:
Now that titles are extracted from the `h1` in the `renderedContent`, the consumers of the
title had to be moved to after the `postProcessHtml` processor.

Closes #17048

mary-poppins commented:
The angular.io preview for 5b91ed88fbe238b8050fa4d86055272b7c3c0b45 is available [here][1].

[1]: https://pr17091-5b91ed88fbe238b8050fa4d86055272b7c3c0b45.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 59c0d3cf9b4e933b15d8964dd2fcc5a8d57a44e1 is available [here][1].

[1]: https://pr17091-59c0d3cf9b4e933b15d8964dd2fcc5a8d57a44e1.ngbuilds.io/

gkalpak commented:
Why can't it find the `Upgraading...` guide 😟:
https://pr17091-59c0d3cf9b4e933b15d8964dd2fcc5a8d57a44e1.ngbuilds.io/?search=upgrad

gkalpak commented on aio/tools/transforms/angular-base-package/processors/convertToJson.js:
```diff
@@ -14,15 +14,11 @@ module.exports = function convertToJsonProcessor(log, createDocMessage) {
 
           // We do allow an empty `title` but resort to `name` if it is not even defined
```
Update comment.

gkalpak commented on aio/tools/transforms/angular-base-package/processors/generateKeywords.js:
```diff
@@ -105,7 +105,7 @@ module.exports = function generateKeywordsProcessor(log, readFilesProcessor) {
 
         doc.searchTerms = {
-          titleWords: extractTitleWords(doc.title || doc.name),
+          titleWords: extractTitleWords(doc.title || doc.name || doc.vFile && doc.vFile.title),
```
Shouldn't `vFile.title` take precedence over name?

gkalpak commented on aio/tools/transforms/angular-base-package/processors/generateKeywords.js:
```diff
@@ -121,10 +121,10 @@ module.exports = function generateKeywordsProcessor(log, readFilesProcessor) {
       docs.push({
         docType: 'json-doc',
         id: 'search-data-json',
-        template: 'json-doc.template.json',
```
Why?

petebacondarwin replied:
We now generate the keywords file **after** the template rendering has completed, so

a) there is no point in providing a template
b) we have to create the `renderedContent` manually

But for b) this is quite easy since it is just a JSON file anyway.

petebacondarwin commented:
This is still broken. Sorry

On Tue, 30 May 2017, 09:39 George Kalpakas, <notifications@github.com>
wrote:

> *@gkalpak* commented on this pull request.
> ------------------------------
>
> In aio/tools/transforms/angular-base-package/processors/convertToJson.js
> <https://github.com/angular/angular/pull/17091#discussion_r119019599>:
>
> > @@ -14,15 +14,11 @@ module.exports = function convertToJsonProcessor(log, createDocMessage) {
>
>            // We do allow an empty `title` but resort to `name` if it is not even defined
>
> Update comment.
> ------------------------------
>
> In
> aio/tools/transforms/angular-base-package/processors/generateKeywords.js
> <https://github.com/angular/angular/pull/17091#discussion_r119020283>:
>
> > @@ -105,7 +105,7 @@ module.exports = function generateKeywordsProcessor(log, readFilesProcessor) {
>
>
>          doc.searchTerms = {
> -          titleWords: extractTitleWords(doc.title || doc.name),
> +          titleWords: extractTitleWords(doc.title || doc.name || doc.vFile && doc.vFile.title),
>
> Shouldn't vFile.title take precedence over name?
> ------------------------------
>
> In
> aio/tools/transforms/angular-base-package/processors/generateKeywords.js
> <https://github.com/angular/angular/pull/17091#discussion_r119020485>:
>
> > @@ -121,10 +121,10 @@ module.exports = function generateKeywordsProcessor(log, readFilesProcessor) {
>        docs.push({
>          docType: 'json-doc',
>          id: 'search-data-json',
> -        template: 'json-doc.template.json',
>
> Why?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/17091#pullrequestreview-40860604>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA9J641wi4Jw6f8MaoXteQPnw84Fww4ks5r-7mlgaJpZM4NpgKO>
> .
>

petebacondarwin commented:
This PR is not quite ready.  Sorry

petebacondarwin committed:
build(aio): capture the h1 title and attach it to the document

The HTML post-processing now collects any h1
that is found in the renderedContent and attaches
it to the doc via the `doc.vFile.title` property.

petebacondarwin committed:
build(aio): use the captured h1 as the title if necessary

If the doc does not already have a `title` tag
then use the one captured from the renderedContent in
the final JSON output, instead.

petebacondarwin committed:
build(aio): the the captured h1 as the title for the search index

If there is no title already provided, use the one captured from the renderedContent.

mary-poppins commented:
The angular.io preview for 4adcc867a9e26775fb96ffc2af1ed92f481691f7 is available [here][1].

[1]: https://pr17091-4adcc867a9e26775fb96ffc2af1ed92f481691f7.ngbuilds.io/

petebacondarwin commented:
OK, this is now working, even for `upgrad...`: https://pr17091-4adcc867a9e26775fb96ffc2af1ed92f481691f7.ngbuilds.io/?search=upgrad

gkalpak reviewed (approved):
LGTM ✨
